L1Trigger/L1THGCal/interface/be_algorithms/HGCalTriggerCellCalibration.h

```diff
@@ -0,0 +1,40 @@
+#ifndef __L1Trigger_L1THGCal_HGCalTriggerCellCalibration_h__
+#define __L1Trigger_L1THGCal_HGCalTriggerCellCalibration_h__
+
+#include <stdint.h>
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "Geometry/HGCalGeometry/interface/HGCalGeometry.h"
+#include "Geometry/Records/interface/IdealGeometryRecord.h"
+#include "Geometry/HGCalCommonData/interface/HGCalDDDConstants.h"
+
+#include "DataFormats/ForwardDetId/interface/HGCEEDetId.h"
+#include "DataFormats/ForwardDetId/interface/HGCHEDetId.h"
+#include "DataFormats/ForwardDetId/interface/HGCalDetId.h"
+#include "DataFormats/L1THGCal/interface/HGCalTriggerCell.h"
+
+class HGCalTriggerCellCalibration{
+
+public:
+
+    HGCalTriggerCellCalibration(const edm::ParameterSet& conf);    
+ 
+    l1t::HGCalTriggerCell calibTrgCell(l1t::HGCalTriggerCell&, const edm::EventSetup& es); 
```

Comment:
Hi Jean-Baptiste, I can change the name into calibrate(). To answer question number 2), I though to retrun a trigger cell in order to put the results of the calibration procedure directly inside the new trigger cell collection, setting, the 4P of the trigger cell directly inside de calibration member function, but sure I agree that it is not the best thing parsing the event setup and retrive the geometry for each cell.. I will try to optimize it